compiler/rustc_middle/src/traits/mod.rs

```diff
@@ -779,14 +795,18 @@ impl DynCompatibilityViolation {
             DynCompatibilityViolation::GAT(name, _) => {
                 format!("it contains the generic associated type `{name}`").into()
             }
+            DynCompatibilityViolation::AsyncFnTrait { .. } => {
+                "`async` function traits are not yet `dyn`-compatible".into()
```

Comment:
```suggestion
                "`async` function traits are not yet dyn-compatible".into()
```

Reply:
What is the reason for preferring backticks for `async` and other keywords, but not for `dyn`? Should I also remove the backticks for `async`?

Reply:
There isn't a clear-cut general answer I'd say. As for *dyn-compatible* specifically, I'd like to think of it as a well-established term — including its spelling — as per https://github.com/rust-lang/lang-team/issues/286 and all PRs under https://github.com/rust-lang/rust/issues/130852.

Reply:
(Personally speaking, I think I'm leaning towards fully avoiding the use of backticks in running text / prose unless it surrounds literal source code for legibility. Mixing "code" and "prose" to express a certain concept feels rather ad hoc and awkward to me — like I'm doing that too when writing GH comments but for "official text" I think that's a different story)

Reply:
https://github.com/rust-lang/lang-team/issues/286 also does not hyphenate. That proposal uses "dyn compatibility" instead of "dyn-compatibility", so to be consistent I suppose I'll use "dyn compatible" rather than "dyn-compatible".

Reply:
> I'll use "dyn compatible" rather than "dyn-compatible".

Ye, that's fine. CC https://github.com/rust-lang/rust/pull/126554#discussion_r1785505496

Reply:
Oh geeze-- yeah, there are lots of uses of "dyn-compatible" throughout the codebase. It would be nice to have a standard here. "dyn compatible" + "dyn-compatibility"? I've added a commit that does a broad search+replace, but we can drop that if you don't think it's justified here.

Reply:
I think it should be done elsewhere

Reply:
Sorry, by elsewhere I mean on a different PR. And then probably assign that to @fmease if they want it. This PR now touches 150 files 😅

Reply:
I'm happy to leave it for another PR. In that case, I'd prefer to keep this PR using the existing terminology that is present throughout the codebase ("dyn-comapatible") and a different PR can sort out whether that choice should be revised. I prefer this approach because it keeps the terminology more uniform in the meantime.